From GKE CI:
```
20:53:38  Step 1/14 : FROM docker.io/library/golang:1.14.2 as builder
20:53:38   ---> 2421885b04da
20:53:38  Step 2/14 : LABEL maintainer="maintainer@cilium.io"
20:53:38   ---> Using cache
20:53:38   ---> 8e0caf011e36
20:53:38  Step 3/14 : ADD . /go/src/github.com/cilium/cilium
20:54:38  failed to export image: failed to set parent sha256:8e0caf011e36f6dd297230873e7f61a2d1f6a560db4e1584f905339950408ffb: unknown parent image ID sha256:8e0caf011e36f6dd297230873e7f61a2d1f6a560db4e1584f905339950408ffb
20:54:38  Makefile:253: recipe for target 'docker-operator-image' failed
20:54:38  make: *** [docker-operator-image] Error 1
```

--- a/Documentation/cmdref/cilium-agent.md
+++ b/Documentation/cmdref/cilium-agent.md
@@ -145,6 +145,7 @@ cilium-agent [flags]
       --mtu int                                       Overwrite auto-detected MTU of underlying network
       --nat46-range string                            IPv6 prefix to map IPv4 addresses to (default "0:0:0:0:0:FFFF::/96")
       --node-port-acceleration string                 BPF NodePort acceleration via XDP ("native", "none") (default "none")
+      --node-port-bind-protection                     Reject application bind(2) requests to service ports in the NodePort range (default true)
       --node-port-mode string                         BPF NodePort mode ("snat", "dsr", "hybrid") (default "snat")
       --node-port-range strings                       Set the min/max NodePort port range (default [30000,32767])
       --policy-audit-mode                             Enable policy audit (non-drop) mode

--- a/Documentation/gettingstarted/kubeproxy-free.rst
+++ b/Documentation/gettingstarted/kubeproxy-free.rst
@@ -356,8 +356,8 @@ the ``cilium status`` CLI command:
     kubectl exec -it -n kube-system cilium-xxxxx -- cilium status | grep KubeProxyReplacement
     KubeProxyReplacement:   Strict   [NodePort (SNAT, 30000-32767, XDP: NATIVE), HostPort, ExternalIPs, HostReachableServices (TCP, UDP)]
 
-NodePort Device and Range
-*************************
+NodePort Device, Port and Bind settings
+***************************************
 
 When running Cilium's BPF kube-proxy replacement, by default, a NodePort or
 ExternalIPs service will be accessible through the IP address of a native device
@@ -365,7 +365,7 @@ which has the default route on the host. To change the device, set its name in
 the ``global.nodePort.device`` helm option.
 
 In addition, thanks to the :ref:`host-services` feature, the NodePort service can
-be accessed by default from a host or a Pod within a cluster via it's public,
+be accessed by default from a host or a pod within a cluster via it's public,
 cilium_host device or loopback address, e.g. ``127.0.0.1:NODE_PORT``.
 
 If ``kube-apiserver`` was configured to use a non-default NodePort port range,
@@ -379,6 +379,14 @@ the reserved ports (``net.ipv4.ip_local_reserved_ports``). This is needed to
 prevent a NodePort service from hijacking traffic of a host local application
 which source port matches the service port. To disable the modification of
 the reserved ports, set ``global.nodePort.autoProtectPortRanges`` to ``false``.
+
+By default, the NodePort implementation prevents application ``bind(2)`` requests
+to NodePort service ports. In such case, the application will typically see a
+``bind: Operation not permitted`` error. This happens either globally for older
+kernels or starting from v5.7 kernels only for the host namespace by default
+and therefore not affecting any application pod ``bind(2)`` requests anymore. In
+order to opt-out from this behavior in general, this setting can be changed for
+expert users by switching ``global.nodePort.bindProtection`` to ``false``.
 
 Container hostPort support
 **************************

--- a/install/kubernetes/cilium/charts/config/templates/configmap.yaml
+++ b/install/kubernetes/cilium/charts/config/templates/configmap.yaml
@@ -336,6 +336,7 @@ data:
 {{- if .Values.global.nodePort.acceleration }}
   node-port-acceleration: {{ .Values.global.nodePort.acceleration | quote }}
 {{- end }}
+  node-port-bind-protection: {{ .Values.global.nodePort.bindProtection | quote }}
   enable-auto-protect-node-port-range: {{ .Values.global.nodePort.autoProtectPortRange | quote }}
 {{- end }}
 

--- a/install/kubernetes/cilium/values.yaml
+++ b/install/kubernetes/cilium/values.yaml
@@ -296,6 +296,10 @@ global:
     # acceleration is the option to accelerate NodePort via XDP
     # acceleration:
 
+    # bindProtection is the option to enable or disable prevention of
+    # applications binding to service ports
+    bindProtection: true
+
     # Append NodePort range to ip_local_reserved_ports if clash with ephemeral
     # ports is detected
     autoProtectPortRange: true

--- a/install/kubernetes/quick-install.yaml
+++ b/install/kubernetes/quick-install.yaml
@@ -151,6 +151,7 @@ data:
   install-iptables-rules: "true"
   auto-direct-node-routes: "false"
   kube-proxy-replacement:  "probe"
+  node-port-bind-protection: "true"
   enable-auto-protect-node-port-range: "true"
   # Chaining mode is set to portmap, enable health checking
   enable-endpoint-health-checking: "true"

